lib/metadata/ocsp/pem_checker.rb

```diff
@@ -10,21 +10,25 @@ def initialize(certificate_factory: Certificate::CertificateFactory.new, ocsp_ch
         @ocsp_checker = ocsp_checker
       end
 
-      def check_pems(pems, ca_files)
+      def check_pems(pems, ca_files, allow_self_signed)
         ca_certs = ca_files.map { |file| OpenSSL::X509::Certificate.new(File.read(file)) }
         issuer_repository = CertificateRepository.new(ca_certs)
         pems.each.with_object({}) do |pem, results|
-          results[pem] = check_pem(pem, issuer_repository)
+          results[pem] = check_pem(pem, issuer_repository, allow_self_signed)
         end
       end
 
-      def check_pem(pem, issuer_repository)
+      def check_pem(pem, issuer_repository, allow_self_signed)
         cert = @certificate_factory.from_inline_pem(pem)
-        chain = issuer_repository.find_chain(cert)
-        store = chain.store
-        issuer = chain.first
-        raise "An issuer was not found in the ca certs for #{cert.subject.to_s}" if issuer.nil?
-        @ocsp_checker.check([cert], issuer, store)[cert]
+        if (allow_self_signed && cert.subject == cert.issuer) then
```

Comment:
Is this the right place to be checking for self-signed certs? I feel like an OCSP checker should just check OCSP and maybe we could filter out self-signed certs at a higher level?